galaxy/importer/loaders/base.py

```diff
@@ -166,14 +95,13 @@ def lint(self):
 
         return all_linters_ok
 
-    def _on_lint_issue(self, linter_type, rule_id, message):
-        extra = {
-            'is_linter_rule_violation': True,
-            'linter_type': linter_type,
-            'linter_rule_id': rule_id,
-            'rule_desc': message
-        }
-        self.log.warning(message, extra=extra)
+    def score(self):
+        return None
+
+    def _on_lint_issue(self, linter_type, rule_id, rule_desc, message=None):
+        lint_record = lintutils.LintRecord(linter_type, rule_id, rule_desc)
```

Comment:
Looks good to me.

(The comments below are about a related idea not directly related to this pr, but making a note while it is in my head...)

I might be tempted in the future to try to batch up the lint result log message logging / ImportTaskMessage creationion a bit, but there would be a premature optimization until its needed. Afaict, since the ImportTaskMessage table doesn't have hard deps on other tables there shouldn't be any issue with creating a lot of them and/or at high rate (ie, there doesn't appear to be any reason at the moment why batching would be needed)

Could the LintRecord creation and the `self.log.warning(message, extra={'lint_record': lint_record})` happen in separate steps?  One method to create/populate the LintRecord and return it, and then send it to the logger elsewhere?

If those are separate, I'm wondering if in the `lint()` method, if all of the LintRecords could be accumulated and sent in a batch as each step finishes (or as smaller batches within each step). ie, wait for N number of LintRecords() to be created, then batch emit them to the logger. _I don't think the style of batching I mention is a necessary optimization currently and may well never be_ (but, again, just making notes while viewing the related code)

Reply:
> Could the LintRecord creation and the self.log.warning(message, extra={'lint_record': lint_record}) happen in separate steps? One method to create/populate the LintRecord and return it, and then send it to the logger elsewhere?

Yes, it's a good idea. I was thinking on the same approach. The main goal was to unblock collections implementations, so it may not be perfect. Let's update it in a separate patch.